### PR TITLE
fixing the bug with squeezing a singleton dimension in torch.min and torch.max

### PR DESCRIPTION
**Pull request by aosokin**
Fixing the bug with squeezing a singleton dimension in torch.min and torch.max in torch.min and torch.max
Resolving #2317

### CODE REVIEW

**Comment by fmassa** [comment]
Could you also add a test for this case? 
Also, does the current code work on the GPU in this case? After quickly glancing at the cuda code, I think it should be ok, just checking.

**Comment by aosokin** [comment]
@fmassa GPU code worked fine.
Added a test, is it in the correct place?

**Comment by gchanan** [comment]
I was just writing a comment saying to put the test there :).

Can you check if test_cuda runs that code?  If not, there are a number of examples where we run the same code from test_cuda; usually passing in a conversion function as a lambda expression.

File: test/test_torch.py
Comment (gchanan) [inline test/test_torch.py]: you can get rid of this line, right?  It's already defined above.

File: test/test_torch.py
Comment (gchanan) [inline test/test_torch.py]: same here.  One issue might be that this function (and the above function) take in a "t" but use "x", so you have to redefine.  Can you fix that above and remove this?

**Comment by aosokin** [comment]
@gchanan fixed, and test_cuda did run that code

**Comment by gchanan** [comment]
This looks good, but it only runs the CPU test in test_cuda (maybe we should be changing the default tensor type or something in test_cuda to avoid this?), and there's a similar problem to the one you found with Mode on GPU.

I'll update this PR soon with a fix for those issues (I'd just do a separate PR but I'm not sure of the right way to merge this and keep TH/THC in sync, so I'll leave it to @soumith).

**Review by gchanan — APPROVED** [review]

**Comment by gchanan** [comment]
@soumith can you merge this when you get a chance?  (involves lib TH/THC changes)

**Comment by soumith** [comment]
merged into master. thanks Anton!

**Pull request closed without merging**
